src/Rule/Trait/PreValidateTrait.php

```diff
@@ -27,10 +27,18 @@ private function preValidate(
             return true;
         }
 
-        if ($rule->shouldSkipOnError() && $context->getParameter($this->parameterPreviousRulesErrored) === true) {
+        if (
+            $rule instanceof SkipOnErrorInterface
+            && $rule->shouldSkipOnError()
+            && $context->getParameter($this->parameterPreviousRulesErrored) === true
+        ) {
             return true;
         }
 
-        return is_callable($rule->getWhen()) && !($rule->getWhen())($value, $context);
+        if ($rule instanceof WhenInterface) {
+            return ($rule->getWhen() !== null) && !($rule->getWhen())($value, $context);
```

Comment:
```suggestion
            $when = $rule->getWhen();

            return $when !== null && !$when($value, $context);
```

Reply:
OR at least the first pair of parentheses can be omitted.

Reply:
Done